src/queryExecutionContext/pipelinedQueryExecutionContext.ts

```diff
@@ -63,10 +60,17 @@ export class PipelinedQueryExecutionContext implements IExecutionContext {
       this.endpoint = new AggregateEndpointComponent(this.endpoint, aggregates);
     }
 
-    // If top then add that to the pipeline
+    // If top then add that to the pipeline. TOP N is effectively LIMIT N OFFSET 0
     const top = partitionedQueryExecutionInfo.queryInfo.top;
     if (typeof top === "number") {
-      this.endpoint = new TopEndpointComponent(this.endpoint, top);
+      this.endpoint = new OffsetLimitEndpointComponent(this.endpoint, top, 0);
+    }
```

Comment:
Do we need error handling cases for these?

Reply:
I don't think so. If we're getting back an error happy to bubble up to the user